diff --git a/include/config.h b/include/config.h
index 92b376d3..beb4361c 100644
--- a/include/config.h
+++ b/include/config.h
@@ -338,7 +338,7 @@ typedef long glyph_t;
 #define SINKS		/* Kitchen sinks - Janet Walz */
 /* dungeon levels */
 #define WALLIFIED_MAZE	/* Fancy mazes - Jean-Christophe Collet */
-#define REINCARNATION	/* Special Rogue-like levels */
+//#define REINCARNATION	/* Special Rogue-like levels */
 /* monsters & objects */
 /*define KOPS*/		/* Keystone Kops by Scott R. Turner */
 	/*They Kops have been removed and replaced with the Keter Sephiroth*/
diff --git a/include/dungeon.h b/include/dungeon.h
index 247ef4e3..9b1d6d6f 100644
--- a/include/dungeon.h
+++ b/include/dungeon.h
@@ -212,7 +212,9 @@ typedef struct branch {
 #define Is_wiz3_level(x)	(on_level(x, &wiz3_level))
 #define Is_sanctum(x)		(on_level(x, &sanctum_level))
 #define Is_portal_level(x)	(on_level(x, &portal_level))
+#ifdef REINCARNATION
 #define Is_rogue_level(x)	(on_level(x, &rogue_level))
+#endif
 #define Is_stronghold(x)	(on_level(x, &stronghold_level))
 #define Is_bigroom(x)		(on_level(x, &bigroom_level))
 #define Is_qstart(x)		(on_level(x, &qstart_level))
diff --git a/src/do_name.c b/src/do_name.c
index 1025af25..139d497d 100644
--- a/src/do_name.c
+++ b/src/do_name.c
@@ -484,7 +484,11 @@ void do_floorname() {
         /* increased chance for fake monster */
         unames[3] = unames[2];
         /* traditional */
+#ifdef REINCARNATION
         unames[4] = roguename();
+#else
+        unames[4] = unames[2];
+#endif
         /* silly */
         unames[5] = "Wibbly Wobbly";
         pline("%s %s to call you \"%s.\"",
diff --git a/src/makemon.c b/src/makemon.c
index b8cbd9cd..c4b1f4ed 100644
--- a/src/makemon.c
+++ b/src/makemon.c
@@ -12833,7 +12833,10 @@ boolean randmonst;
 			mkmon_template = MOLY_TEMPLATE;
 		}
 		/* most general case at bottom -- creatures randomly being zombified */
-		else if(randmonst && can_undead(ptr) && !Is_rogue_level(&u.uz)
+		else if(randmonst && can_undead(ptr)
+#ifdef REINCARNATION
+			&& !Is_rogue_level(&u.uz)
+#endif
 			&& !Infuture
 		){
 			if(In_mines(&u.uz)){
@@ -14896,9 +14899,10 @@ rndmonst()
 			}
 		}
 	}
+#ifdef REINCARNATION
 	if(Is_rogue_level(&u.uz))
 		return roguemonst();
-
+#endif
 	if (rndmonst_state.choice_count < 0) {	/* need to recalculate */
 	    int zlevel, minmlev, maxmlev;
 	    boolean elemlevel;
diff --git a/src/mon.c b/src/mon.c
index 4e9b0cce..2587ea7c 100644
--- a/src/mon.c
+++ b/src/mon.c
@@ -3775,7 +3775,11 @@ struct monst * mdef;	/* another monster which is next to it */
 	}
 	
 	/* undead vs civs */
-	if(!(In_cha(&u.uz) || Is_rogue_level(&u.uz))){
+	if(!(In_cha(&u.uz)
+#ifdef REINCARNATION
+		|| Is_rogue_level(&u.uz)
+#endif
+	)){
 		if(!(magr->mpeaceful && mdef->mpeaceful && is_undead(youracedata))){
 			if(mm_undead(magr) && 
 				(!is_witch_mon(mdef) && mdef->mtyp != PM_WITCH_S_FAMILIAR && !mdef->mpetitioner && !mm_undead(mdef) && !mindless_mon(mdef) && mdef->mfaction != YELLOW_FACTION)
diff --git a/util/dgn_comp.y b/util/dgn_comp.y
index 13a6a5a6..6251c4cd 100644
--- a/util/dgn_comp.y
+++ b/util/dgn_comp.y
@@ -494,7 +494,7 @@ getchain(s)
 
 	if(strlen(s)) {
 
-	    for(i = n_levs - tmpdungeon[n_dgns].levels + 1; i <= n_levs; i++)
+	    for(i = n_levs - tmpdungeon[n_dgns].levels; i <= n_levs; i++)
 		if(!strcmp(tmplevel[i].name, s)) return i;
 
 	    yyerror("Can't locate the specified chain level.");
